### ISSUE TITLE
SQLite3 configuration missed WAL mode

### ISSUE THREAD

**Issue by dukhovnyi** [amplitude/Amplitude-iOS#509]
## Expected Behavior
No performance warnings appears.

## Current Behavior
Xcode throws warning `Reading or writing to a database not configured in WAL mode will perform excessive I/O and reduce the health of storage devices.` when Amplitude uses `sqlite3_step` interface.

![image](https://github.com/user-attachments/assets/d18a5301-a599-441f-abc5-c7599b3f023e)

## Possible Solution
I suppose the `SQLString` during `sqlite3_exec` in [file](https://github.com/amplitude/Amplitude-iOS/blob/ca2bf8e6c8581dc181156f250e19a75fffef3d7c/Sources/Amplitude/AMPDatabaseHelper.m#L231) should be extended with `PRAGMA journal_mode=WAL;` to turn on WAL mode.
Consequences of this change should be tracked and tested ...

## Steps to Reproduce
1. Install 8.22.0 using Swift Package Manager into existed project.
2. Build project.
3. Observe warnings.

## Environment
- SDK Version: 8.22
- Device: iPhone 12+
- OS Version: 18.x
- Xcode: 16.1

**Comment by elychkouski** [comment]
Up!

ChatGPT suggests such improvements:

Add the pragma immediately after the sqlite3_open succeeds, before any statements run:
```objective-c
if (sqlite3_open([_databasePath UTF8String], &_database) == SQLITE_OK) {
    // One call is enough – SQLite remembers this on disk.
    sqlite3_exec(_database, "PRAGMA journal_mode=WAL;", NULL, NULL, NULL);
    // Optional: trims fsyncs to a single flush.
    sqlite3_exec(_database, "PRAGMA synchronous=NORMAL;", NULL, NULL, NULL);
}
```
Enabling SQLite write‑ahead logging (WAL) for every database you open eliminates the warning and greatly reduces disk traffic. The message appears because your code (and many third‑party SDKs that still default to the older rollback journal) opens the file in the legacy DELETE journal mode, which forces an extra full‑file write on every commit; iOS 17/18’s runtime diagnostics flag this as “excessive I/O.” ￼ ￼

Because the rollback journal rewrites entire pages, even light traffic can hit >5 MB/s on an SSD‑backed simulator or device, which triggers the runtime check added in Xcode 15.2+. Apple’s “Reducing Disk Writes” guide now explicitly recommends WAL for SQLite files. ￼ ￼

**Bottom line:** add PRAGMA journal_mode=WAL; (and optionally PRAGMA synchronous=NORMAL;) right after opening every SQLite connection. That single line silences the warning, cuts battery‑draining write amplification, and usually speeds up your database access as a bonus.

**Comment by crleona** [comment]
This SDK is in maintenance mode and is only receiving critical updates. Please migrate to our Swift SDK: https://amplitude.com/docs/sdks/analytics/ios/ios-sdk-migration-guide.
